src/Approximations/decompositions.jl

```diff
@@ -659,3 +659,11 @@ function project(H::AbstractHyperrectangle, block::AbstractVector{Int})
     πr = radius_hyperrectangle(H)[block]
     return Hyperrectangle(πc, πr, check_bounds=false)
 end
+
+function project(V::Union{<:VPolygon{N}, <:VPolytope{N}},
+                 block::AbstractVector{Int}) where {N}
+    n = dim(V)
+    M = projection_matrix(block, n, N)
+    πvertices = broadcast(v -> M * v, vertices_list(V))
+    return basetype(V)(πvertices)
```

Comment:
Projecting a `VPolygon` would probably give you an `Interval`. You need to check the dimension.

Reply:
I suggest to exclude the `VPolygon` case here. The implementation can be made more efficiently if you know that you go to 1D. In particular, you do not need to create the list of vertices and then run a convex hull. Instead you can keep track of upper and lower bound and in the end return an `Interval`.

Reply:
>  I am surprised that this does not crash.

VPolygon constructor doesn't check dimension (unless you pass a vertices matrix). I think it's a good idea to return an interval if the output dimension of `M` is `1`. we could also return a VPolytope.

Reply:
For now we can return a `VPolytope` and open an issue.

Reply:
> I suggest to exclude the VPolygon case here.

yes, though the implementation in this pr is better than the fallback implementation in master i guess? (which converts to constraints list). that can be handled as a follow up issue?

Reply:
The current implementation is wrong. If you change it to `VPolytope` that is fine.